Project-Starlord/Controllers/AccountController.cs

```diff
@@ -161,6 +165,95 @@ public async Task<ActionResult<UserModel>> Register(UserModel userModel)
             return userModel;
         }
 
+        [HttpPost]
+        [Route("ForgotPassword")]
+        public async Task<ActionResult<IActionResult>> ForgotPassword(string email)
+        {
+            if (String.IsNullOrWhiteSpace(email))
+            {
+                return BadRequest();
+            }
+
+            if (_context.Users.Any(e => e.Email == email))
+            {
+                ResetTokenModel model = new ResetTokenModel
+                {
+                    Email = email,
+                    Expires = DateTime.UtcNow.AddHours(24),
+                    IsTokenUsed = false
+                };
+
+                // Generate reset token
+                byte[] resetTokenBytes;
+                new RNGCryptoServiceProvider().GetBytes(resetTokenBytes = new byte[16]);
+                model.Token = Convert.ToBase64String(resetTokenBytes);
+
+                //todo: send reset email or whatever
+                // i really don't know what to do here..
+                string message =
+                    "Hi! To reset your password you need to open the following link within 24h: \n\n https://wanderer.app/passwordReset?token=" +
+                    model.Token;
```

Comment:
@CJVisser So, how are we going to do this?

Reply:
Just let it rest for now. Fill fix later on